addons/hw_drivers/iot_handlers/drivers/PrinterDriver.py

```diff
@@ -208,7 +209,55 @@ def format_escpos(self, im):
             raster_data += raster_send + width.to_bytes(2, 'little') + slice_height.to_bytes(2, 'little') + im_slice
             dots = dots[width*max_slice_height:]
 
-        return raster_data + RECEIPT_PRINTER_COMMANDS['escpos']['cut']
+        return raster_data
+
+    def _int_low_high(self, inp_number):
+        res = b''
+        for _ in range(2):
+            inp_number, r = divmod(inp_number, 256)
+            res += r.to_bytes(1, 'little')
+        return res
+
+    def format_escpos_bit_image_column(self, im):
+        """ prints with the `ESC *`-command """
+
+        def extract_columns_from_picture(im):
+            # In high density, the number of bits for vertical data = 24
+            width_pixels, height_pixels = im.size
+            LINE_HEIGHT = 24
+            TOP = 0
+            for left in range(0, width_pixels, LINE_HEIGHT):
+                box = (left, TOP, left + LINE_HEIGHT, TOP + height_pixels)
+                slice = im.transform(
+                    (LINE_HEIGHT, height_pixels),
+                    Image.EXTENT,
+                    box
+                )
+                yield slice.tobytes()
+
+        # escpos ESC * command print column per column
+        # (instead of usual row by row).
+        # So we transpose the picture to ease the calculations
+        im = im.transpose(Image.ROTATE_270).transpose(Image.FLIP_LEFT_RIGHT)
+
+        ESC = b"\x1b"
+        HEADER = ESC + b'*!' + self._int_low_high(im.height)
+
+        raster_data = ESC + b'3\x10'  # Adjust line-feed size
+        for column in extract_columns_from_picture(im):
+            raster_data += HEADER + column + b"\n"
+        raster_data += ESC + b'2'  # Reset line-feed size
+        return raster_data
+
+    def format_escpos(self, im):
+        # GS v 0 incompatible with TM-U2x0 models according to:
+        # https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=94
+        must_use_column = "TM-U2" in self.device_name
```

Comment:
indeed, please re-read my message above:

> Also, I was willing to have a way for the user to control if they would prefer to use ESC or GV. I thought maybe a particular suffix at the end of the printer name (like "-force-raster") in case new model of printer that does not support GV are used. I tried to modify the name of the printer using IoT cups, but on odoo service restart, the printer name lose the modified name to get back it's previous name.
What do you think ?
Little reminder: GV is obsolete so will likely not be featured in new printers of the brand, see:
https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=94


> Another way to know if a printer need to use column to print ?

In theory they can all used it, but as it is slower and the quality is bad, the questions is rather do we have a way to know if GS v 0 is not supported. On this i don't think there is a way to know except checking the table of compatibility.
This was the reason why I was willing to make it "an option" so that the client have a way to force the column printing if necessary (see previous paragraph)